sync_check/config.go

```diff
@@ -21,64 +21,88 @@ import (
 	"github.com/juju/errors"
 	"github.com/ngaut/log"
 	"github.com/pingcap/tidb-tools/sync_check/util"
+	"github.com/pingcap/tidb/model"
 )
 
 const (
 	percent0   = 0
 	percent100 = 100
 )
 
-// NewConfig creates a new config.
-func NewConfig() *Config {
-	cfg := &Config{}
-	cfg.FlagSet = flag.NewFlagSet("sync-check", flag.ContinueOnError)
-	fs := cfg.FlagSet
-
-	fs.StringVar(&cfg.ConfigFile, "config", "", "Config file")
-	fs.StringVar(&cfg.LogLevel, "L", "info", "log level: debug, info, warn, error, fatal")
-	fs.IntVar(&cfg.Delay, "delay", 0, "check data 5 second before")
-	fs.IntVar(&cfg.ChunkSize, "chunk-size", 1000, "diff check chunk size")
-	fs.IntVar(&cfg.Sample, "sample", 100, "the percent of sampling check")
-	fs.IntVar(&cfg.CheckThCount, "check-thcount", 1, "the count of check thread count")
-	fs.StringVar(&cfg.TimeField, "time-field", "", "a field with datetime type")
-	fs.StringVar(&cfg.BeginTime, "begin-time", "", "check data's begin time")
-	fs.StringVar(&cfg.EndTime, "end-time", "", "check data's end time")
-	fs.StringVar(&cfg.SplitField, "split-field", "", "use this field split data to several chunk")
-	fs.BoolVar(&cfg.UseRowID, "use-rowid", false, "set true if target-db and source-db all support tidb implicit column _tidb_rowid")
-	return cfg
+// TableCheckCfg is the config of table to be checked.
+type TableCheckCfg struct {
+	// table name
+	Name   string `toml:"name"`
+	// Schema is seted in SourceDBCfg
+	Schema string
+	// field should be the primary key, unique key or field with index
```

Comment:
if I dont set unique key or primary key, I would get a wrong result, right?

Reply:
I think `checkConfig` should check it

Reply:
this field just used to split data, not order by key.

Reply:
if not set, find a suitable index?